### ISSUE TITLE
Nick-Highlight

### ISSUE THREAD

**Issue by demlak** [poljar/weechat-matrix#181]
maybe stupid question.. but.. how to highlight someone?
in riot webclient i just start typing `@usern..` and i get the username to choose

what do i have to write-down to highlight someone via matrix protocol?

thx

**Comment by poljar** [comment]
There is no way to do this currently, the highlights in matrix are called pills and are HTML links.

The input parser would need to be able to parse markdown links, which I find unergonomic, or we would need to have some custom syntax for pills/mentions.

**Comment by trygveaa** [comment]
Wouldn't it be best to just turn `@nick` into a pill?

**Comment by poljar** [comment]
Yeah I think I thought about that syntax as well, thought still care needs to be taken to not pill silly things (check if the user is in the member list), and support this when tab completion is used.

**Comment by makoshark** [comment]
Turning `@nick` into a pill seems like an obvious (and important!) feature since the ability to create notifications/highlights for others in channels seems pretty basic functionality. Requiring the `@` seems reasonable and I think it's how wee-slack does it (I guess @trygveaa would know?).

It's a small thing this issue is the only place I found this limitation documented. I'd love for this issue to be marked as a feature  request with a more descriptive title "turn nick mentions into pills to allow weechat-matrix users to create highlights for matrix others" or something?

If nobody else is going to make progress on this I might find time to take a stab although I'm completely new to any of the related codebases or systems so it will probably take me some time and effort.

**Comment by trygveaa** [comment]
> Requiring the @ seems reasonable and I think it's how wee-slack does it (I guess @trygveaa would know?).

Yeah, wee-slack makes it a Slack mention if you write a nick with `@` prefixed. If you just write the nick it will just be regular text. That's how it works in the official Slack clients, so made sense to do the same in wee-slack. It looks like Element for Matrix also behaves this way, so I think it would make sense to do the same in weechat-matrix.
